core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -52,6 +52,16 @@ private[spark] class CoarseGrainedExecutorBackend(
     driver ! RegisterExecutor(executorId, hostPort, cores)
     context.system.eventStream.subscribe(self, classOf[RemotingLifecycleEvent])
   }
+  
+  override def isValidMessage(msg: Any) = msg match {
+    case disEvt: DisassociatedEvent =>
+      if (disEvt.remoteAddress == driver.anchorPath.address) {
```

Comment:
Ah sorry about it. In this case, I think it actually makes more sense to revert to your original approach.